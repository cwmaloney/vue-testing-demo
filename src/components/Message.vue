<template>
  <h1 class="message"> {{ message || 'no message' }}</h1>
</template>

<script>
export default {
  name: 'message',

  props: {
    'message': { type: String }
  }
}
</script>

<style lang="css" scoped>

.message {
  color: blue;
}

</style>
