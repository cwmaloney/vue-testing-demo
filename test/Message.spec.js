import { shallowMount } from '@vue/test-utils'
import Message from '../src/components/Message.vue'

describe('Message.vue', () => {

  it('display the message defined in props', () => {
    const message = 'test message';
    const wrapper = shallowMount(Message, {
      propsData: { message }
    });
    expect(wrapper.text()).toBe(message);
  });

  it('displays default message if no messages defined', () => {
    const defaultMessage = 'no message';
    const wrapper = shallowMount(Message);
    expect(wrapper.text()).toBe(defaultMessage);
  });

  it('rendors message', () => {
    const message = 'a test message';
    const wrapper = shallowMount(Message, {
      propsData: { message }
    });
   expect(wrapper.html()).toMatchSnapshot();
 });

});
