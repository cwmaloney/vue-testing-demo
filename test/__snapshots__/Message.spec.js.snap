// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Message.vue rendors message 1`] = `<h1 class="message"> a test message</h1>`;
